interfaces/builtin/serial_port.go

```diff
@@ -208,6 +210,64 @@ func (iface *serialPortInterface) AutoConnect(*snap.PlugInfo, *snap.SlotInfo) bo
 	return true
 }
 
+func (iface *serialPortInterface) HotplugDeviceDetected(di *hotplug.HotplugDeviceInfo) (*hotplug.ProposedSlot, error) {
+	bus, _ := di.Attribute("ID_BUS")
+	if di.Subsystem() != "tty" || bus != "usb" || !serialDeviceNodePattern.MatchString(di.DeviceName()) {
+		return nil, nil
+	}
+
+	slot := hotplug.ProposedSlot{
+		Attrs: map[string]interface{}{
+			"path": di.DeviceName(),
+		},
+	}
+	if vendor, ok := di.Attribute("ID_VENDOR_ID"); ok {
+		slot.Attrs["usb-vendor"] = vendor
+	}
+	if product, ok := di.Attribute("ID_MODEL_ID"); ok {
+		slot.Attrs["usb-product"] = product
+	}
+	return &slot, nil
+}
+
+func compareSlotAndDeviceAttribute(slotAttributeValue int64, di *hotplug.HotplugDeviceInfo, devinfoAttribute string) bool {
```

Comment:
Nitpick: can you please reorder the arguments and tweak the name:
```suggestion
func slotDeviceAttrEqual(di *hotplug.HotplugDeviceInfo, devinfoAttribute string, slotAttributeValue int64) bool {
```

It would make more sense, at least in my eyes.